File: .ci/doc/templates/print-search-result.tpl.java
import io.kuzzle.sdk.Kuzzle;
import java.util.*;
import io.kuzzle.sdk.protocol.*;
import io.kuzzle.sdk.coreClasses.responses.Response;
import io.kuzzle.sdk.coreClasses.SearchResult;
import io.kuzzle.sdk.coreClasses.lang.Lang;

public class SnippetTest {
  private static Kuzzle kuzzle;

  public static void main(String[] argv) {
    try {
      AbstractProtocol protocol;
      if (System.getenv("SNIPPET_PROTOCOL") != null && System.getenv("SNIPPET_PROTOCOL").equals("http")) {
        protocol = new Http("kuzzle");
      } else {
        protocol = new WebSocket("kuzzle");
      }
      kuzzle = new Kuzzle(protocol);
      kuzzle.connect();
      [snippet-code]
      System.out.println("Successfully retrieved " + matched.size() + " documents");
    } catch (Exception e) {
      e.printStackTrace();
    } finally {
      if (kuzzle != null) {
        kuzzle.disconnect();
      }
    }
  }
}

File: .ci/doc/templates/print-search-result.tpl.kt
val protocol: AbstractProtocol;
if (System.getenv("SNIPPET_PROTOCOL") != null && System.getenv("SNIPPET_PROTOCOL") == "http") {
  protocol = Http("kuzzle")
} else {
  protocol = WebSocket("kuzzle")
}
val kuzzle = Kuzzle(protocol).apply {
  connect()
}
try {
  [snippet-code]
  println("Successfully retrieved " + matched.size + " documents");  } catch (e: Exception) {
  e.printStackTrace()
} finally {
  kuzzle.disconnect()
}